rest-api-guidelines/quality/evolution/naming-conventions.md

```diff
@@ -13,6 +13,14 @@ Every identifier **MUST** be in American English and written in `lowercase`. An
 
 Every URI **MUST** follow the General Rules except for the `camelCase` rule. Instead, a hyphen \(`-`\) **SHOULD** be used to delimit combined words \(kebab-case\). Besides, a URI **MUST NOT** end with a trailing slash \(`/`\).
 
+Plural nouns SHOULD be used in the URI where appropriate to identify collections of data resources (e.g. ``/orders``, ``/products``).
+
+An individual resource in a collection of resources MAY exist directly beneath the collection URI.  (e.g. ``/orders/{order_id}``).
+
+Sub-resource collections MAY exist directly beneath an individual resource. This should convey a relationship to another collection of resources (e.g. ``/orders/{order_id}/items``).
```

Comment:
In general, I a not a fan of policing URL structures and inferring any semantics from the URL structure - e.g. "A is s a sub-resource of B because I am guessing it from the URL of `/A/B`". 

Per RFC 7320: 

https://tools.ietf.org/html/rfc7320

> The most successful strategy for applications and extensions that wish to use URIs is to use them in the fashion they were designed: as links that are exchanged as part of the protocol, rather than statically specified syntax.

With that being said, I see a good intention in this PR and as such, I am hesitant to refuse it, but also to merge it. Let's have some discusson about it :)

Reply:
the intention of this PR comes from the fact, that I see a lot of APIs which have a semantically incoherent structure and/or have endpoints like /member/bulk, etc. And yes, I agree, that it may suggest a parent/child structure in some cases. On the other hand how would you express a relations between some resources?

Reply:
Are we good if I delete the sub-resources part and leave only the part on individual resource in a collection @zdne ?

Reply:
Yes please.

Reply:
done. Can I merge it?